using-packages.md

```diff
@@ -0,0 +1,262 @@
+Note however that many packages depend on other frameworks, such as the
+[AngularDart](https://webdev.dartlang.org/angular/) web framework, and those
+will not work with Flutter. We hope to offer a better search experience with
+support for filtering to only packages that are compatible with Flutter in the
+future.
+
+{% include note.html content="Flutter is still a young platform, and only a
+fairly small set of Flutter packages are currently available on pub. We
+encourage all Flutter developers to [develop](/developing-packages/) new
+packages." %}
+
+### Adding a package dependency to an app
+
+To add a plugin 'plugin1' to an app:
+
+1. Open the `pubspec.yaml` file located inside your app folder, and add
+`plugin1:` under `dependencies`.
+
+1. Get the plugin:
+   * From the terminal: Run `flutter packages get`
+   * From IntelliJ: Click 'Packages Get' in the action ribbon at the top of `pubspec.yaml`
+
+1. Build or run your app. As part of this, Flutter will add the package code to
+   your app.
+
+## Managing package dependencies & versions
+
+### Package versions
+
+All packages have a version number, specified in their `pubspec.yaml` file. Pub
+displays the current version of a package next to it's name (for example, see
+the [url_launcher](https://pub.dartlang.org/packages/url_launcher) package), as
+well as a list of all prior versions ([url_launcher
+versions](https://pub.dartlang.org/packages/url_launcher#pub-pkg-tab-versions)).
+
+When a package is added to `pubspec.yaml` using the shorthand form `plugin1:`
+this is interpreted as `plugin1: any`, i.e. any version of the package may be
+used. To ensure your app does not break when a package is updated, we recommend
+specifying a version range using one of the following formats:
+
+* Range constraints: Specify a minimum and maximim version, e.g.:
+```dart
+dependencies:
+    url_launcher: '>=0.1.2 <0.2.0'
+```
+
+* Range constraint with [*caret syntax*](https://www.dartlang.org/tools/pub/dependencies#caret-syntax): 
+Similar to a regular range constraints
+```dart
+dependencies:
+    collection: '^0.1.2'
+```
+
+For additional details, see the [Pub versioning guide](https://www.dartlang.org/tools/pub/versioning).
+
+### Updating package dependencies
+
+When you run `flutter packages get` ('Packages Get' in IntelliJ) for the first
+time after adding a package, Flutter saves the concrete package version found in
+the `pubspec.lock`
+[lockfile](https://www.dartlang.org/tools/pub/glossary#lockfile). This ensures
+that you get the same version again if you, or another developer on your team,
+run `flutter packages get`.
+
+If you want to upgrade to a new version of the package, for example to use new
+features in that package, run `flutter packages upgrade` ('Upgrade dependencies'
+in IntelliJ). This will retrive the highest available version of the package,
+which is allowed by the version constraint you have specified in `pubspec.yaml`.
+
+### Dependencies on unpublished packages
```

Comment:
This seems generic and duplicates existing docs. Should we just link to somewhere?